MobileWallet/Screens/Profile/ProfileViewController.swift

```diff
@@ -0,0 +1,188 @@
+        let middleLabelText = NSLocalizedString("Transacting in person? Your sender can get your emoji ID by scanning this QR code in their Tari app", comment: "Profile middle label")
+
+        self.middleLabel.text = middleLabelText
+        self.middleLabel.font = Theme.shared.fonts.profileMiddleLabel
+        self.middleLabel.textColor = Theme.shared.colors.profileMiddleLabel!
+    }
+
+    private func genQRCode() {
+
+        guard let wallet = TariLib.shared.tariWallet else {
+            return
+        }
+
+        let (walletPublicKey, pubKeyError) = wallet.publicKey
+        if pubKeyError != nil {
+            UserFeedback.shared.error(
+                title: NSLocalizedString("Failed to access wallet public key", comment: "Profile View Controller"), description: ""
+            )
+        }
+        let qrText = walletPublicKey?.emojis.0 ?? ""
+        let vcard = qrText.data(using: .utf8)
+
+        let filter = CIFilter(name: "CIQRCodeGenerator")
+        filter?.setValue(vcard, forKey: "inputMessage")
+        filter?.setValue("L", forKey: "inputCorrectionLevel")
+
+        if let output = filter?.outputImage {
+            let scaleX = qrImageView.frame.size.width / output.extent.size.width
+            let scaleY = qrImageView.frame.size.height / output.extent.size.height
+            let transform = CGAffineTransform(scaleX: scaleX, y: scaleY)
+            let scaledOutput = output.transformed(by: transform)
+            qrImageView.image = UIImage(ciImage: scaledOutput)
+        }
+
+        qrContainer.layer.shadowOpacity = 1
+        qrContainer.layer.shadowOffset = CGSize(width: 20, height: 20)
+        qrContainer.layer.shadowRadius = 3.0
+        qrContainer.layer.shadowColor = Theme.shared.colors.profileQRShadow?.cgColor
+
+        let shadowRect: CGRect = qrContainer.bounds.insetBy(dx: 4, dy: 4)
+        qrContainer.layer.shadowPath = UIBezierPath(rect: shadowRect).cgPath
+        qrContainer.layer.shouldRasterize = true
+        qrContainer.layer.rasterizationScale = UIScreen.main.scale
+        qrContainer.layer.masksToBounds = false
+    }
+
+    private func customizeViews() {
+        customizeTitleLabel()
+        customizeEmoji()
+        customizeCopyMyEmojiButton()
+        customizeSeparatorView()
+        customizeMiddleLabel()
+        genQRCode()
+    }
+
+    private func copyToClipboard() {
+        let pasteboard = UIPasteboard.general
+        pasteboard.string = emojiLabel.text ?? ""
+    }
+
+    private func sendHapticNotification() {
+        let impactFeedbackgenerator = UIImpactFeedbackGenerator(style: .light)
+        impactFeedbackgenerator.impactOccurred()
+    }
+
+    // MARK: - Actions
+    @IBAction func onDismissAction(_ sender: Any) {
+        self.dismiss(animated: true, completion: nil)
+    }
+
+    @IBAction func onCopyEmojiAction(_ sender: Any) {
+        copyToClipboard()
+        sendHapticNotification()
+
+        UIView.transition(with: copyEmojiButton,
```

Comment:
This doesn't look right when I tried it. It's in a faded transition state for 2 seconds and then it jumps back. I think just keep this one simple.
Change the text to `Copied!` right away, then use `DispatchQueue.main.asyncAfter(deadline: .now() + 2)` to change it back after 2 seconds